src/components/structures/MessagePanel.js

```diff
@@ -286,6 +287,47 @@ module.exports = React.createClass({
 
             var last = (i == lastShownEventIndex);
 
+            var isMembershipChange = (e) =>
```

Comment:
suggest hoisting this out of the `for` loop so that it doesn't need to be redefined for each event